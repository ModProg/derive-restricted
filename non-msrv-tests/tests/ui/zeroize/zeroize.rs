extern crate zeroize_ as zeroize;

use core::marker::PhantomData;

use derive_where::DeriveWhere;

#[derive(DeriveWhere)]
#[derive_where(Zeroize(test); T)]
struct InvalidOption<T, U>(T, PhantomData<U>);

#[derive(DeriveWhere)]
#[derive_where(Zeroize(test = "test"); T)]
struct WrongOptionSyntax1<T, U>(T, PhantomData<U>);

#[derive(DeriveWhere)]
#[derive_where(Zeroize("option"); T)]
struct WrongOptionSyntax2<T, U>(T, PhantomData<U>);

#[derive(DeriveWhere)]
#[derive_where(Zeroize(crate(zeroize_)); T)]
struct WrongCrateSyntax<T, U>(T, PhantomData<U>);

#[derive(DeriveWhere)]
#[derive_where(Zeroize(crate = "struct Test"); T)]
struct InvalidCrate<T, U>(T, PhantomData<U>);

#[derive(DeriveWhere)]
#[derive_where(Zeroize(drop, drop); T)]
struct DuplicateDrop<T, U>(T, PhantomData<U>);

#[derive(DeriveWhere)]
#[derive_where(Zeroize(crate = "zeroize_", crate = "zeroize_"); T)]
struct DuplicateCrate<T, U>(T, PhantomData<U>);

#[derive(DeriveWhere)]
#[derive_where(Zeroize(drop, drop, crate = "zeroize_"); T)]
struct DuplicateDropWithCrate<T, U>(T, PhantomData<U>);

#[derive(DeriveWhere)]
#[derive_where(Zeroize(drop, crate = "zeroize_", crate = "zeroize_"); T)]
struct DropWithDuplicateCrate<T, U>(T, PhantomData<U>);

#[derive(DeriveWhere)]
#[derive_where(Zeroize(crate = "zeroize_", crate = "zeroize_", drop); T)]
struct DuplicateCrateWithDrop<T, U>(T, PhantomData<U>);

#[derive(DeriveWhere)]
#[derive_where(Zeroize(crate = "zeroize_", drop, drop); T)]
struct CrateWithDuplicateDrop<T, U>(T, PhantomData<U>);

fn main() {}
