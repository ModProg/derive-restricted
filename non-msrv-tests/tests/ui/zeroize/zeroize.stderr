error: `Zeroize` doesn't support this option
 --> tests/ui/zeroize/zeroize.rs:5:24
  |
5 | #[derive_where(Zeroize(test))]
  |                        ^^^^

error: `Zeroize` doesn't support this option
 --> tests/ui/zeroize/zeroize.rs:8:24
  |
8 | #[derive_where(Zeroize(test = "test"))]
  |                        ^^^^

error: unexpected option syntax
  --> tests/ui/zeroize/zeroize.rs:11:24
   |
11 | #[derive_where(Zeroize("option"))]
   |                        ^^^^^^^^

error: unexpected option syntax
  --> tests/ui/zeroize/zeroize.rs:14:24
   |
14 | #[derive_where(Zeroize(crate(zeroize_)))]
   |                        ^^^^^^^^^^^^^^^

error: expected path, expected identifier
  --> tests/ui/zeroize/zeroize.rs:17:32
   |
17 | #[derive_where(Zeroize(crate = "struct Test"))]
   |                                ^^^^^^^^^^^^^

error: duplicate `crate` option
  --> tests/ui/zeroize/zeroize.rs:20:44
   |
20 | #[derive_where(Zeroize(crate = "zeroize_", crate = "zeroize_"))]
   |                                            ^^^^^^^^^^^^^^^^^^
